kubernetes/Dockerfile

```diff
@@ -1,6 +1,6 @@
 FROM alpine
 
-ARG VAULT_VERSION=1.13.2
+ARG VAULT_VERSION=1.16.3
```

Comment:
bump to latest community release on this occasion:
* https://github.com/hashicorp/vault/blob/main/CHANGELOG.md#1163